maths/series/arithmetico_geometric_sequence.py

```diff
@@ -12,134 +12,198 @@
-    '(2.0 + 9.0) x 4.0x0.5^3', '(2.0 + 12.0) x 4.0x0.5^4']
-    Value Of Last Term : 3.50
-    Sum Of Your A.G.S. : 35.00
-    Infinite Series Sum : 40.00
-    Value Of Which Term You Want : 2
-    Value Of 2-th Term : 10.00
+    def __init__(
+        self,
+        arithmetic_value: float,
+        arithmetic_difference: float,
+        geometric_value: float,
+        common_ratio: float,
+        term_count: int,
+    ) -> None:
+        self.arithmetic_value = arithmetic_value
+        self.arithmetic_difference = arithmetic_difference
+        self.geometric_value = geometric_value
+        self.common_ratio = common_ratio
+        self.term_count = term_count
+
+    """:params arithmetic_value: Initial Value For Arithmetic Progression
+        arithmetic_difference: Difference For Arithmetic Progression
+        geometric_value: Initial Value For Geometric Progression
+        common_ratio: Common Ratio For Geometric Progression
+        term_count: Number Of Terms
+        k_in_kth_term: Value Of K In K-th Term
     """
 
-    def full_series(self):
+    def full_series(self) -> list:
+        """
+        >>> AGS(2, 3, 5, 6, 3).full_series()
+        ['2 x 5', '(2 + 3) x 5x6^1', '(2 + 6) x 5x6^2']
+        >>> AGS(3, 2.1, 1.3, 0.4, 3).full_series()
+        ['3 x 1.3', '(3 + 2.1) x 1.3x0.4^1', '(3 + 4.2) x 1.3x0.4^2']
+        >>> AGS(9, 3, 2, -0.5, 0).full_series()
+        []
+        """
         series = []
-        for i in range(self.n):
+        for i in range(self.term_count):
             if i == 0:
-                series.append(str(self.a) + " x " + str(self.b))
+                series.append(
+                    str(self.arithmetic_value) + " x " + str(self.geometric_value)
+                )
             else:
-                ap_part = str(self.a) + " + " + str(i * self.d)
-                gp_part = str(self.b) + "x" + str(self.r) + "^" + str(i)
-                series.append("(" + ap_part + ")" + " x " + gp_part)
+                ap_part = (
+                    str(self.arithmetic_value)
+                    + " + "
+                    + str(i * self.arithmetic_difference)
+                )
+                gp_part = (
+                    str(self.geometric_value)
+                    + "x"
+                    + str(self.common_ratio)
+                    + "^"
+                    + str(i)
+                )
+                series.append(f"({ap_part}) x {gp_part}")
         return series
 
-    def last_term_value(self):
-        return (self.a + (self.n - 1) * self.d) * (self.b * pow(self.r, self.n - 1))
-
-    def sum(self):
+    def last_term_value(self) -> float:
+        """
+        >>> AGS(2, 3, 5, 6, 3).last_term_value()
+        1440
+        >>> AGS(3, 2.1, 1.3, 0.4, 3).last_term_value()
+        1.4976000000000003
+        >>> AGS(9, 3, 2, -0.5, 0).last_term_value()
+        'None'
+        """
+        if self.term_count == 0:
+            return "None"
```

Comment:
The type hint promises that this function returns a float, not a str.  If we can not calculate a float to return then we could raise a ValueError.